### PR TITLE
fix: scenario where unselecting the first page and importing fails

### LINKED ISSUE #17023: [Bug]: Templates which have hidden pages crashes the app when forked
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/67054171/191939917-e26ab120-3c6f-472e-8bd5-d6250f14cec5.mov

When we click on `ADD PAGES` we call `api/v1/app-templates/621c6c6589659f362e07f7b3/merge/632d846bff5c311c00c83fd9/632d7205ff5c311c00c83e06` POST with payload as an array of the page names selected. [`"Dashboard", "Search"]`

The response returns an application with 4 pages(showing only the pages returned here).

One of them is the default pages that is in the app already. Two other pages are the pages we selected while forking the template. The fourth page with id as `Selected` is a hidden page. The problem here is `Selected` is actually the name of the page and not the page id. 

```
[
  {
    "id": "632d8895ff5c311c00c8403c",
    "isDefault": false,
    "defaultPageId": "632d8895ff5c311c00c8403c",
    "default": false
  },
  {
    "id": "632d8895ff5c311c00c8403a",
    "isDefault": false,
    "defaultPageId": "632d8895ff5c311c00c8403a",
    "default": false
  },
  {
    "id": "Selected",
    "isDefault": false,
    "default": false
  },
  {
    "id": "632d881dff5c311c00c84036",
    "isDefault": true,
    "defaultPageId": "632d881dff5c311c00c84036",
    "default": true
  }
]
```

Once we get this list of pages the frontend then tries to get all the actions, js collections etc of the pages using the page id. So we end up calling `api/v1/actions?pageId=Selected` GET to get the list of actions but it returns a 404 since `Selected` is not the page id. The frontend shows the error page when it fails to fetch the list of actions in the current page.

The expectation is to return the correct id for the hidden page instead of the page name in the id field.

### Steps To Reproduce

Steps are the same as shown in the screen recording.

### Public Sample App

_No response_

### Version

Cloud / Release - v1.8.4

### LINKED ISSUE #17270: [Bug]: On forking selected pages from a template, resource not found error is shown
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On forking "Response" page of Employee time tracker and "Search" Page of Customer Support Dashboard, resource not found is shown, and user is not directed to forked page, however on hard refresh the redirection happens
Screenshot 2022-10-03 at 3 39 11 PM

 

### Steps To Reproduce

1. Click on start from template card
2. Add Search page from Customer Support Dashboard
3. Observe the resource not found error is thrown

### Public Sample App

_No response_

### Version

Release

### LINKED ISSUE #17272: [Bug]: Forking selected pages from template from Add new page popover is crashing the app
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On forking selected pages from Add new page popover, crashes the app with page not found error

Screenshot 2022-10-03 at 4 03 00 PM

### Steps To Reproduce

1. Click on + on left pane of Pages
2. Select Add Page from Template option
3. Select 'Search' page of Customer Support Dashboard template 
Note : Happening on unchecking first page, and selecting other pages.
5. Observe the app crashes

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

Fixes #17023 
Fixes #17270
Fixes #17272

Fixes a scenario which was missed by https://github.com/appsmithorg/appsmith/pull/17038. That PR was related to not making use of the pages returned by the response of the fork template api as they were incorrect. This PR removes the usage of the same for navigation. Instead to  get the page id we make use of the get all pages selector.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually. Tests are being added in a different PR https://github.com/appsmithorg/appsmith/pull/17036

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/sagas/TemplatesSagas.ts [hunk 1]
```diff
@@ -241,17 +238,8 @@ function* forkTemplateToApplicationSaga(
...
-      });
-      const newPages = differenceBy(
-        postImportPageList,
-        currentListOfPages,
-        "pageId",
-      );
-
-      // Fetch the actions/jsobjects of the new set of pages that have been added
       yield call(postPageAdditionSaga, applicationId);
+      const pages: string[] = yield select(getAllPageIds);
```
Comment (eco-monk) [inline app/client/src/sagas/TemplatesSagas.ts:242]: nit: I suggest renaming `pages` to `pageIds`

**Review by eco-monk — COMMENTED** [review]
Do we need to handle this @`importTemplateToWorkspaceSaga` too?

We're using the `application` object (from importTemplate API response) to fetch `defaultPageId`.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**